### PR TITLE
many: verify that unit tests work with nosecboot tag and without secboot package

### PR DESCRIPTION
**Pull request by bboozzoo**
Some distros, eg. Debian Sid, run unit tests as part of the package build process. However, when dependencies are not vendored, there is no secboot package, and thus the unit tests fail to build. We can disable secboot by passing the `nosecboot`, package, but unit tests still fail to build and work. The branch attempts to fix that and add a step to the unit tests workflow that ensures we keep it working.

This should also fix #9502.

### CODE REVIEW

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -168,6 +168,15 @@ jobs:
       run: |
         cd ${{ github.workspace }}/src/github.com/snapcore/snapd || exit 1
         ./run-checks --unit
+    - name: Test Go (nosecboot)
```
Comment (zyga) [inline .github/workflows/test.yaml:171]: Really nice!

File: cmd/snap-bootstrap/cmd_initramfs_mounts.go [hunk 2]
```diff
@@ -74,9 +73,9 @@ var (
 		snap.TypeSnapd:  "snapd",
 	}
 
-	secbootMeasureSnapSystemEpochWhenPossible = secboot.MeasureSnapSystemEpochWhenPossible
-	secbootMeasureSnapModelWhenPossible       = secboot.MeasureSnapModelWhenPossible
-	secbootUnlockVolumeIfEncrypted            = secboot.UnlockVolumeIfEncrypted
+	secbootMeasureSnapSystemEpochWhenPossible func() error
```
Comment (zyga) [inline cmd/snap-bootstrap/cmd_initramfs_mounts.go:76]: A comment about the indirection would be useful, I think.

File: run-checks [hunk 3]
```diff
@@ -1,4 +1,4 @@
-#!/bin/sh -eu
+#!/bin/bash -eu
```
Comment (zyga) [inline run-checks:1]: I'm feeling lucky ;-)

**Review by zyga — APPROVED** [review]
Looks very good.

File: run-checks [hunk 4]
```diff
@@ -292,14 +293,20 @@ if [ "$UNIT" = 1 ]; then
     command -v go
     go version
 
+    tags=()
```
Comment (mvo5) [inline run-checks:296]: This is a bit hard to read but it seems like it's just the way it is with shell :/ Slightly sad that there seems to be no better way. Thanks for finding a solution for this!

**Review by mvo5 — APPROVED** [review]
Thank you

File: run-checks [hunk 5]
```diff
@@ -292,27 +293,36 @@ if [ "$UNIT" = 1 ]; then
     command -v go
     go version
 
+    tags=
```
Comment (mvo5) [inline run-checks:296]: This works? Nice trick!

**Pull request merged**